### PR TITLE
Add a comment with wrong type should not be possible

### PR DESCRIPTION
**Pull request by juherr**
Hi,

I am able to create comments without a good type, check: http://retroachievements.org/searchresults.php?s=test+juherr

I didn't note their id and I'm not able to remove them. Could someone do it for me?

BTW, here a PR that will block such kind of adds.

### CODE REVIEW

File: src/ArticleType.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+namespace RA;
+
+abstract class ArticleType
+{
+    public const Game = 1;
+
+    public const Achievement = 2;
+
+    public const User = 3;
+
```
Comment (juherr) [inline src/ArticleType.php:12]: I didn't find any usage of the value 4 but it may exist.

Comment (ScottFromDerby) [reply src/ArticleType.php:12]: It probably used to exist... I think maybe it was "News"? I'm not sure, sorry

File: lib/database/activity.php [hunk 2]
```diff
@@ -307,6 +307,10 @@ function RemoveComment($articleID, $commentID, $userID, $permissions)
 
 function addArticleComment($user, $articleType, $articleID, $commentPayload, $onBehalfOfUser = null)
 {
+    if (!in_array($articleType, \RA\ArticleType::values())) {
```
Comment (ScottFromDerby) [inline lib/database/activity.php:310]: Love this

Comment (juherr) [reply lib/database/activity.php:310]: Thank! Anothers options are: 
* `\RA\ArticleType::isValid($articleType)`, or
* directly use https://github.com/myclabs/php-enum (but I don't know what is the meaning here about 3rd party lib)

File: src/ArticleType.php (hunk 1 above)
Comment (juherr) [reply src/ArticleType.php:12]: Good catch! After a 2nd look, I found that the enum already exists under `SubjectType` where the value 4 was defined.
I've just reworked my PR in order to merge the 2 enums.
Thank for the review :D

**Comment by juherr** [comment]
@luchaos 
The idea behind the migration file is just going ahead: try to clean the inconsistent data by a baby step and stop future inconsistencies table by table.

For performance, you can add indexes if they are not yet in place. It will have the same result.

I've no idea how big is the table and how much error exists, then how much will be the effort for more consistency.
But I think it should be done a day ;)

Feel free to pick the commits you prefer, they are supposed to work alone and separate for that purpose.

**Comment by luchaos** [comment]
Fantastic, thanks. 
This reminds me; I still owe you an answer on the framework/v2 issue - most of my time goes into a complete rewrite of both code and data structure. Proper indexes and foreign key constraints will be in place there. That's one of the reasons why I haven't taken the time yet to go forward with cleanup tasks. That said, v1 maintenance is very much relevant until then - fixing improper behaviour like in this PR is _very_ welcome

**Pull request merged**